File: copy_css_verify.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');
const crypto = require('crypto');

const SOURCE = "/home/jessessh/htdocs/srv1113343.hstgr.cloud/app/design/frontend/AWA_Custom/ayo_home5_child/web/css";
const DESTA = "/home/user/htdocs/srv1113343.hstgr.cloud/app/design/frontend/AWA_Custom/ayo_home5_child/web/css";
const DESTB = "/home/user/htdocs/srv1113343.hstgr.cloud/pub/static/frontend/AWA_Custom/ayo_home5_child/pt_BR/css";

const FILES = [
  "awa-global-components.css",
  "awa-bugfixes-v2.css",
  "awa-extra-pages.css",
  "awa-checkout-cart.css",
  "awa-customer-pages.css",
  "awa-cms-404.css",
  "awa-mobile-polish.css",
  "awa-animations-a11y.css",
  "awa-final-polish.css",
  "b2b/shoppinglist.css",
];

function computeSha256(filePath) {
  const hash = crypto.createHash('sha256');
  const data = fs.readFileSync(filePath);
  hash.update(data);
  return hash.digest('hex');
}

console.log("=" . repeat(100));
console.log("STEP 1: CREATE DIRECTORIES");
console.log("=" .repeat(100));

try {
  fs.mkdirSync(path.join(DESTA, "b2b"), { recursive: true });
  console.log(`✓ Created: ${DESTA}/b2b`);
} catch (e) {
  console.log(`✗ Error: ${e.message}`);
}

try {
  fs.mkdirSync(path.join(DESTB, "b2b"), { recursive: true });
  console.log(`✓ Created: ${DESTB}/b2b`);
} catch (e) {
  console.log(`✗ Error: ${e.message}`);
}
console.log("");

console.log("=" .repeat(100));
console.log("STEP 2: COPY FILES SOURCE -> DESTA");
console.log("=" .repeat(100));

for (const file of FILES) {
  try {
    const src = path.join(SOURCE, file);
    const dst = path.join(DESTA, file);
    fs.mkdirSync(path.dirname(dst), { recursive: true });
    fs.copyFileSync(src, dst);
    console.log(`✓ ${file}`);
  } catch (e) {
    console.log(`✗ ${file}: ${e.message}`);
  }
}
console.log("");

console.log("=" .repeat(100));
console.log("STEP 3: COPY FILES SOURCE -> DESTB");
console.log("=" .repeat(100));

for (const file of FILES) {
  try {
    const src = path.join(SOURCE, file);
    const dst = path.join(DESTB, file);
    fs.mkdirSync(path.dirname(dst), { recursive: true });
    fs.copyFileSync(src, dst);
    console.log(`✓ ${file}`);
  } catch (e) {
    console.log(`✗ ${file}: ${e.message}`);
  }
}
console.log("");

console.log("=" .repeat(100));
console.log("STEP 4: COMPUTE SHA-256 HASHES AND VERIFICATION");
console.log("=" .repeat(100));
console.log("");

console.log("| File | Source Hash (first 16) | DestA Hash (first 16) | DestB Hash (first 16) | MatchA | MatchB |");
console.log("|---|---|---|---|---|---|");

let allMatch = true;
const hashes = [];

for (const file of FILES) {
  try {
    const srcPath = path.join(SOURCE, file);
    const destaPath = path.join(DESTA, file);
    const destbPath = path.join(DESTB, file);

    const srcHash = computeSha256(srcPath);
    const destaHash = computeSha256(destaPath);
    const destbHash = computeSha256(destbPath);

    const matchA = srcHash === destaHash ? "✓" : "✗ MISMATCH";
    const matchB = srcHash === destbHash ? "✓" : "✗ MISMATCH";

    if (srcHash !== destaHash || srcHash !== destbHash) {
      allMatch = false;
    }

    const srcShort = srcHash.substring(0, 16) + "...";
    const destaShort = destaHash.substring(0, 16) + "...";
    const destbShort = destbHash.substring(0, 16) + "...";

    console.log(`| ${file} | ${srcShort} | ${destaShort} | ${destbShort} | ${matchA} | ${matchB} |`);
    hashes.push([file, srcHash, destaHash, destbHash, matchA, matchB]);
  } catch (e) {
    console.log(`| ${file} | ERROR | ERROR | ERROR | ERROR | ERROR |`);
    allMatch = false;
  }
}

console.log("");
console.log("=" .repeat(100));
console.log("VERIFICATION SUMMARY");
console.log("=" .repeat(100));
console.log(`Source: ${SOURCE}`);
console.log(`DestA:  ${DESTA}`);
console.log(`DestB:  ${DESTB}`);
console.log("");

if (allMatch) {
  console.log("✓✓✓ ALL FILES VERIFIED - HASHES MATCH ✓✓✓");
} else {
  console.log("✗✗✗ HASH MISMATCH DETECTED - REVIEW REQUIRED ✗✗✗");
}

console.log("");
console.log("=" .repeat(100));
console.log("DETAILED HASH TABLE (FULL SHA-256)");
console.log("=" .repeat(100));
console.log("");
console.log("| File | Source SHA-256 | DestA SHA-256 | DestB SHA-256 | Match |");
console.log("|---|---|---|---|---|");

for (const [file, srcHash, destaHash, destbHash, matchA, matchB] of hashes) {
  const match = (matchA === "✓" && matchB === "✓") ? "✓ BOTH" : "✗ FAIL";
  console.log(`| ${file} | ${srcHash} | ${destaHash} | ${destbHash} | ${match} |`);
}

console.log("");
console.log("=" .repeat(100));
console.log("FILE COUNT AND STATUS");
console.log("=" .repeat(100));
console.log(`Total files processed: ${FILES.length}`);
console.log(`All matches: ${allMatch ? "YES" : "NO"}`);

File: copy_css_verify.py
#!/usr/bin/env python3
"""
Copy CSS files from source to two destinations with SHA-256 verification
Handles both Unix and Windows path styles
"""
import os
import hashlib
import shutil
import sys
from pathlib import Path

# Try to normalize paths - support both Unix /home and Windows C:\Users
def get_path(*parts):
    """Convert path parts to platform-appropriate path"""
    if os.path.exists('/home'):
        # Unix-like environment
        return os.path.join(*parts)
    else:
        # Convert /home/user to C:\Users\user or similar
        path = os.path.join(*parts)
        if path.startswith('/home/'):
            # Map /home/user to C:\Users\user
            home_part = path[6:]  # Remove /home/
            if home_part.startswith('user/'):
                return 'C:\\Users\\user\\' + home_part[5:].replace('/', '\\')
            elif home_part.startswith('jessessh/'):
                return 'C:\\Users\\jessessh\\' + home_part[9:].replace('/', '\\')
        return path

SOURCE = "/home/jessessh/htdocs/srv1113343.hstgr.cloud/app/design/frontend/AWA_Custom/ayo_home5_child/web/css"
DESTA = "/home/user/htdocs/srv1113343.hstgr.cloud/app/design/frontend/AWA_Custom/ayo_home5_child/web/css"
DESTB = "/home/user/htdocs/srv1113343.hstgr.cloud/pub/static/frontend/AWA_Custom/ayo_home5_child/pt_BR/css"

FILES = [
    "awa-global-components.css",
    "awa-bugfixes-v2.css",
    "awa-extra-pages.css",
    "awa-checkout-cart.css",
    "awa-customer-pages.css",
    "awa-cms-404.css",
    "awa-mobile-polish.css",
    "awa-animations-a11y.css",
    "awa-final-polish.css",
    "b2b/shoppinglist.css",
]

def compute_sha256(filepath):
    """Compute SHA-256 hash of a file"""
    try:
        sha256_hash = hashlib.sha256()
        with open(filepath, "rb") as f:
            for byte_block in iter(lambda: f.read(4096), b""):
                sha256_hash.update(byte_block)
        return sha256_hash.hexdigest()
    except Exception as e:
        return f"ERROR: {str(e)}"

print("=" * 100)
print("STEP 1: CREATE DIRECTORIES")
print("=" * 100)
try:
    Path(os.path.join(DESTA, "b2b")).mkdir(parents=True, exist_ok=True)
    print(f"✓ Created: {DESTA}/b2b")
except Exception as e:
    print(f"✗ Error: {e}")

try:
    Path(os.path.join(DESTB, "b2b")).mkdir(parents=True, exist_ok=True)
    print(f"✓ Created: {DESTB}/b2b")
except Exception as e:
    print(f"✗ Error: {e}")
print()

print("=" * 100)
print("STEP 2: COPY FILES SOURCE -> DESTA")
print("=" * 100)
for file in FILES:
    try:
        src = os.path.join(SOURCE, file)
        dst = os.path.join(DESTA, file)
        os.makedirs(os.path.dirname(dst), exist_ok=True)
        shutil.copy2(src, dst)
        print(f"✓ {file}")
    except Exception as e:
        print(f"✗ {file}: {e}")
print()

print("=" * 100)
print("STEP 3: COPY FILES SOURCE -> DESTB")
print("=" * 100)
for file in FILES:
    try:
        src = os.path.join(SOURCE, file)
        dst = os.path.join(DESTB, file)
        os.makedirs(os.path.dirname(dst), exist_ok=True)
        shutil.copy2(src, dst)
        print(f"✓ {file}")
    except Exception as e:
        print(f"✗ {file}: {e}")
print()

print("=" * 100)
print("STEP 4: COMPUTE SHA-256 HASHES AND BUILD VERIFICATION TABLE")
print("=" * 100)
print()
print("| File | Source Hash (first 16) | DestA Hash (first 16) | DestB Hash (first 16) | MatchA | MatchB |")
print("|---|---|---|---|---|---|")

all_match = True
hashes_full = []

for file in FILES:
    try:
        src_path = os.path.join(SOURCE, file)
        desta_path = os.path.join(DESTA, file)
        destb_path = os.path.join(DESTB, file)
        
        src_hash = compute_sha256(src_path)
        desta_hash = compute_sha256(desta_path)
        destb_hash = compute_sha256(destb_path)
        
        match_a = "✓" if src_hash == desta_hash else "✗ MISMATCH"
        match_b = "✓" if src_hash == destb_hash else "✗ MISMATCH"
        
        if src_hash != desta_hash or src_hash != destb_hash:
            all_match = False
        
        src_short = src_hash[:16] + "..." if len(src_hash) > 16 else src_hash
        desta_short = desta_hash[:16] + "..." if len(desta_hash) > 16 else desta_hash
        destb_short = destb_hash[:16] + "..." if len(destb_hash) > 16 else destb_hash
        
        print(f"| {file} | {src_short} | {desta_short} | {destb_short} | {match_a} | {match_b} |")
        hashes_full.append((file, src_hash, desta_hash, destb_hash, match_a, match_b))
    except Exception as e:
        print(f"| {file} | ERROR | ERROR | ERROR | ERROR | ERROR |")
        all_match = False

print()
print("=" * 100)
print("VERIFICATION SUMMARY")
print("=" * 100)
print(f"Source: {SOURCE}")
print(f"DestA:  {DESTA}")
print(f"DestB:  {DESTB}")
print()

if all_match:
    print("✓✓✓ ALL FILES VERIFIED - HASHES MATCH ✓✓✓")
else:
    print("✗✗✗ HASH MISMATCH DETECTED - REVIEW REQUIRED ✗✗✗")

print()
print("=" * 100)
print("DETAILED HASH TABLE (FULL SHA-256)")
print("=" * 100)
print()
print("| File | Source SHA-256 | DestA SHA-256 | DestB SHA-256 | Match |")
print("|---|---|---|---|---|")

for file, src_hash, desta_hash, destb_hash, match_a, match_b in hashes_full:
    match = "✓ BOTH" if (match_a == "✓" and match_b == "✓") else "✗ FAIL"
    print(f"| {file} | {src_hash} | {desta_hash} | {destb_hash} | {match} |")

print()
print("=" * 100)
print("FILE COUNT AND STATUS")
print("=" * 100)
print(f"Total files processed: {len(FILES)}")
print(f"All matches: {'YES' if all_match else 'NO'}")
